apps/src/lib/kits/maker/MBFirmataWrapper.js

```diff
@@ -19,4 +24,19 @@ export default class MicrobitFirmataWrapper extends MBFirmataClient {
   analogRead(pin, callback) {
     callback(this.analogChannel[pin]);
   }
+
+  trackDigitalComponent(pin, callback) {
+    this.digitalCallbacks.push({pin, callback});
+  }
+
+  receivedDigitalUpdate(chan, pinMask) {
+    super.receivedDigitalUpdate(chan, pinMask);
+    for (let i = 0; i < this.digitalCallbacks.length; i++) {
+      let pin = this.digitalCallbacks[i].pin;
+      let callback = this.digitalCallbacks[i].callback;
+      // We add one here to translate from the 0/1 of digital input to the
+      // first part of the 6-array input of button, see ExternalButton.js
```

Comment:
Sorry I'm not really sure I understand this comment? Does this correspond to `['', 'down', 'up', 'click', 'long-click', 'hold']` so like 0 -> 1 for the index of down and 1 -> 2 for the index of up?

Reply:
Yes

Reply:
The button knows "connected/not-connected" so this translates it to the 6-part array of button behavior. Luckily because we only support up and down at the moment, we don't need to do additional logic to figure out if it's a long-click, etc.

Are there changes I can make to make it clearer?

Reply:
Ah I see. I would make the comment more explicit then:
`Translate 0/1 to 1/2 corresponding to ExternalButton.boardEvents[1/2] for 'down'/'up'`